pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -94,13 +135,18 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 					// add to cache if the pod is running or pending
 					pc.update(ip, key)
 				}
+				if _, ok := pc.podTransitionTime[ip]; !ok && IsPodReady(pod) {
+					pc.podTransitionTime[ip] = podCondition.LastTransitionTime
```

Comment:
is k8s using LastTransitionTime to make decisions? I am wondering if this is a reliable mechanism for this

Reply:
Yes, from testing, this works well. I've tested locally with pods that have no readiness and readiness check. And it looks like this is only way to achieve otherwise we need to record and store it at Istio side.